packages/mds-audit/api.ts

```diff
@@ -116,14 +119,27 @@ function api(app: express.Express): express.Express {
   })
 
   /**
-   * Audit middleware to load the audit into locals using the audit_trip_id
+   * Audit middleware to load the audit and latest provider event into locals
    */
   app.use(pathsFor('/trips/:audit_trip_id'), async (req: AuditApiTripRequest, res: AuditApiResponse, next) => {
     try {
       const { audit_trip_id } = req.params
-      if (isValidAuditTripId(audit_trip_id)) {
-        res.locals.audit_trip_id = audit_trip_id
-        res.locals.audit = await readAudit(audit_trip_id)
+      if (!isValidAuditTripId(audit_trip_id)) {
+        return next()
+      }
+      res.locals.audit_trip_id = audit_trip_id
+      res.locals.audit = await readAudit(audit_trip_id)
+      let { provider_id, provider_vehicle_id } = req.body
+      if ((!provider_id || !provider_vehicle_id) && res.locals.audit) {
+        provider_id = res.locals.audit.provider_id
+        provider_vehicle_id = res.locals.audit.provider_vehicle_id
+      }
+      const device = await readDeviceByVehicleId(provider_id, provider_vehicle_id)
```

Comment:
Shouldn't this be calling `getVehicle`? Also, there's a chance it won't be found so you probably check for `undefined` ... also, this middleware `app.use` is called for every request that starts with `/trips/:audit_trip_id` which is probably not what you want. I don't think you really need to do this here, or stash the event in res.locals.